components/portal/timeout/controllers.js

```diff
@@ -54,13 +70,69 @@ define(['angular'], function(angular) {
       }
     }
 
+    /**
+     * check if the session is close to timing out due to inactivity
+     */
+    function checkInactive() {
+      var inactiveDuration = sessionInactivityService.getInactiveDuration();
+
+      var warningInactiveDuration =
+          toMillis(appInactiveTimeout - (2 * checkInactiveDelay));
+      var dialogInactiveDuration =
+          toMillis(appInactiveTimeout - checkInactiveDelay);
+      var finalInactiveDuration =
+          toMillis(appInactiveTimeout);
+
+      if (inactiveDuration > finalInactiveDuration) {
+        redirect();
+      } else if (inactiveDuration > warningInactiveDuration) {
+        // Trigger dialog with one minute on the clock
+        $timeout(triggerDialog,
+            dialogInactiveDuration - inactiveDuration, false);
+      }
+    }
+
+    /**
+     * Reload to continue the session
+     */
+    function reload() {
+      $window.location.reload(true);
+    }
+
+    /**
+     * Redirect to end the session;
+     */
+    function redirect() {
+      $window.location.replace(MISC_URLS.loginURL);
+    }
+
     /**
      * Trigger a Session Expired dialog box
      */
     function triggerDialog() {
       $mdDialog.show({
-        templateUrl:
-          'portal/timeout/session-expired.html',
+        /*
+         * this template is inline to avoid accidentally
+         * auto-renewing the session.
+         */
+        template:
+          '<md-dialog class="dialog__session-expired" ' +
+          'aria-label="are you still there">' +
+          '  <md-toolbar>' +
+          '    <div class="md-toolbar-tools">' +
+          '      <h2>Are you still there?</h2>' +
+          '    </div>' +
+          '  </md-toolbar>' +
+          '  <md-dialog-content class="md-dialog-content">' +
+          '    <p>Your session will expire soon. Please reload the ' +
+          'page to continue, otherwise you will be logged out.</p>' +
+          '  </md-dialog-content>' +
+          '  <md-dialog-actions layout="row">' +
+          '    <md-button aria-label="" ng-click="reloadPage()">' +
+          '      Reload' +
```

Comment:
How feasible might it be to give the user a less invasive action to take than reloading the page outright? An async request that just tickles the server to extend the session timeout? [ssa.gov's guidance on timeout dialogs](https://www.ssa.gov/accessibility/bpl/bps/forms/timeout/default.htm), for instance, talks about advantages in returning the user back to what they were focused on as best one can when the user chooses to extend the session. Reloading the page is potentially going to discard whatever they were in the middle of working on behind this pop-up.

Reply:
Cool. Better is better. Seems to me that this changeset already implements an experience better than the status quo even from a narrow perspective of accessibility. I'd therefore be supportive of a path forward that merges this increment of progress and backlogs the further improvement to keep users where they're at in the application in a re-upped session, if you'd want to go that route.

Reply:
yeah, I'm going to backlog an improvement for this.